core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementService.java

```diff
@@ -0,0 +1,117 @@
+            Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stack.getId());
+            if (!loadBalancers.isEmpty()) {
+                Optional<LoadBalancer> loadBalancerOptional = selectLoadBalancer(loadBalancers);
+                if (loadBalancerOptional.isEmpty()) {
+                    LOGGER.error("Unabled to determine load balancer type. Load balancer will not be registered with FreeIPA.");
+                    return;
+                }
+
+                LoadBalancer loadBalancer = loadBalancerOptional.get();
+                String endpoint = loadBalancer.getEndpoint();
+                String domain = getDomainFromStack(stack);
+
+                if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
+                    AddDnsCnameRecordRequest request = new AddDnsCnameRecordRequest();
+                    request.setCname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setTargetFqdn(StringUtils.appendIfMissing(loadBalancer.getDns(), DOMAIN_PART_DELIMITER));
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsCnameRecord(request);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    AddDnsARecordRequest request = new AddDnsARecordRequest();
+                    request.setHostname(endpoint);
+                    request.setDnsZone(domain);
+                    request.setIp(loadBalancer.getIp());
+                    request.setEnvironmentCrn(stack.getEnvironmentCrn());
+                    dnsV1Endpoint.addDnsARecord(request);
+                } else {
+                    LOGGER.error("Unable to find DNS or IP for load balancer. Load balancer will not be registered with FreeIPA.");
+                }
+            }
+        }
+    }
+
+    public void deleteLoadBalancerDomainFromFreeIPA(Stack stack) {
+        if (manageCertificateAndDnsInPem()) {
+            Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stack.getId());
+            if (!loadBalancers.isEmpty()) {
+                Optional<LoadBalancer> loadBalancerOptional = selectLoadBalancer(loadBalancers);
+                if (loadBalancerOptional.isEmpty()) {
+                    LOGGER.debug("Unabled to determine load balancer type. Load balancer will not be deleted from FreeIPA.");
+                    return;
+                }
+
+                LoadBalancer loadBalancer = loadBalancerOptional.get();
+                String endpoint = loadBalancer.getEndpoint();
+                String domain = getDomainFromStack(stack);
+
+                if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
+                    dnsV1Endpoint.deleteDnsCnameRecord(stack.getEnvironmentCrn(), domain, endpoint);
+                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                    dnsV1Endpoint.deleteDnsARecord(stack.getEnvironmentCrn(), domain, endpoint);
```

Comment:
I think it would be nice to add a log line so we would know from the logs which one was called